contrib/twitchtv/twirp/example_test.go

```diff
@@ -0,0 +1,56 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package twirp_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	twirptrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/twitchtv/twirp"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/twitchtv/twirp/example"
+)
+
+func ExampleWrapClient() {
+	// Start the tracer
+	tracer.Start()
+	defer tracer.Stop()
+
+	client := example.NewHaberdasherJSONClient("http://localhost:8080", twirptrace.WrapClient(&http.Client{}))
+	for i := 0; i < 10; i++ {
+		hat, err := client.MakeHat(context.Background(), &example.Size{Inches: 6})
+		if err != nil {
+			fmt.Println("error making hat:", err)
+			continue
+		}
+		fmt.Println("made hat:", hat)
+	}
+}
+
+type hatmaker struct{}
+
+func (hatmaker) MakeHat(ctx context.Context, size *example.Size) (*example.Hat, error) {
+	hat := &example.Hat{
+		Size: 42,
+		Color: "cornflower blue",
+		Name: "oversized blue hat",
+	}
+	return hat, nil
+}
+
+func ExampleNewServerHooks() {
+	server := example.NewHaberdasherServer(hatmaker{}, twirptrace.NewServerHooks())
+	// Note: it is recommended to use WrapServer in conjunction with NewServerHooks
```

Comment:
Why? Can we clarify? The example is documentation and should be clear with minimal open questions...

Reply:
I'll revisit this.

Reply:
I had a look into this, basically seeing if we could remove `WrapServer` and extract the necessary information from the context passed to callbacks.
This didn't work - the request headers needed to extract a span are only available in the "server" and not in the callback's context.
(I'd hoped that [twirp.WithHTTPRequestHeaders](https://godoc.org/github.com/twitchtv/twirp#WithHTTPRequestHeaders) would get called in the generated service implementation, but it seems to need user code to call it.)

So instead I've removed the example and mention `WrapServer` in the docstring for `NewServerHooks`. The `ExampleWrapServer` remains as a suitable example for server-side tracing.